src/enclave/Enclave/NonObliviousSortMergeJoin.cpp

```diff
@@ -81,10 +80,27 @@ void non_oblivious_sort_merge_join(uint8_t *join_expr, size_t join_expr_length,
   // so last_foreign_row.get() is guaranteed to not be null.
   FlatbuffersTemporaryRow last_foreign_row;
 
+  // A "dummy" row needed for FullOuter joins so that 
+  // any_primary_row.get() is guaranteed to not be null
+  FlatbuffersTemporaryRow any_primary_row;
```

Comment:
It looks like in any case, whenever we write to output using `last_foreign_row` or `any_primary_row`, their fields are all nulled. Could you update these variables to be something like `dummy_foreign_row` and `dummy_primary_row`? And also remove the call to `last_foreign_row.set(current);` when first encountering a new foreign row. I think it would help to improve readability.

Reply:
Sure, will do